src/storage/v2/disk/storage.cpp

```diff
@@ -0,0 +1,1703 @@
+}
+
+VertexAccessor DiskStorage::DiskAccessor::CreateVertex() {
+  auto gid = storage_->vertex_id_.fetch_add(1, std::memory_order_acq_rel);
+  auto acc = vertices_.access();
+
+  auto *delta = CreateDeleteObjectDelta(&transaction_);
+  auto [it, inserted] = acc.insert(Vertex{storage::Gid::FromUint(gid), delta});
+  MG_ASSERT(inserted, "The vertex must be inserted here!");
+  MG_ASSERT(it != acc.end(), "Invalid Vertex accessor!");
+
+  if (delta) {
+    delta->prev.Set(&*it);
+  }
+
+  return {&*it, &transaction_, &storage_->indices_, &storage_->constraints_, config_};
+}
+
+VertexAccessor DiskStorage::DiskAccessor::CreateVertex(utils::SkipList<Vertex>::Accessor &accessor, storage::Gid gid,
+                                                       const std::vector<LabelId> &label_ids,
+                                                       PropertyStore &&properties, Delta *delta) {
+  OOMExceptionEnabler oom_exception;
+  auto *disk_storage = static_cast<DiskStorage *>(storage_);
+  disk_storage->vertex_id_.store(std::max(disk_storage->vertex_id_.load(std::memory_order_acquire), gid.AsUint() + 1),
+                                 std::memory_order_release);
+  auto [it, inserted] = accessor.insert(Vertex{gid, delta});
+  MG_ASSERT(inserted, "The vertex must be inserted here!");
+  MG_ASSERT(it != accessor.end(), "Invalid Vertex accessor!");
+  /// TODO: move
+  for (auto label_id : label_ids) {
+    it->labels.push_back(label_id);
+  }
+  it->properties = std::move(properties);
+  if (delta) {
+    delta->prev.Set(&*it);
+  }
+  return {&*it, &transaction_, &storage_->indices_, &storage_->constraints_, config_};
+}
+
+std::optional<VertexAccessor> DiskStorage::DiskAccessor::FindVertex(storage::Gid gid, View view) {
+  auto acc = vertices_.access();
+  auto vertex_it = acc.find(gid);
+  if (vertex_it != acc.end()) {
+    return VertexAccessor::Create(&*vertex_it, &transaction_, &storage_->indices_, &storage_->constraints_, config_,
+                                  view);
+  }
+  for (const auto &vec : index_storage_) {
+    acc = vec->access();
+    auto index_it = acc.find(gid);
+    if (index_it != acc.end()) {
+      return VertexAccessor::Create(&*index_it, &transaction_, &storage_->indices_, &storage_->constraints_, config_,
+                                    view);
+    }
+  }
+
+  rocksdb::ReadOptions read_opts;
+  auto strTs = utils::StringTimestamp(transaction_.start_timestamp);
+  rocksdb::Slice ts(strTs);
+  read_opts.timestamp = &ts;
+  auto *disk_storage = static_cast<DiskStorage *>(storage_);
+  auto it = std::unique_ptr<rocksdb::Iterator>(
+      disk_transaction_->GetIterator(read_opts, disk_storage->kvstore_->vertex_chandle));
+  for (it->SeekToFirst(); it->Valid(); it->Next()) {
+    const auto &key = it->key();
+    if (Gid::FromUint(std::stoull(utils::ExtractGidFromKey(key.ToString()))) == gid) {
+      return LoadVertexToMainMemoryCache(key, it->value());
+    }
+  }
+  return std::nullopt;
+}
```

Comment:
I think in this function we check whether vertex exists in vertices_ first, then we check whethere there is vertex in index, then we get it from storage and load it to cache. But in loding to cache we again check if vertex exists in skip list and I think this repetition is costly.